### PR TITLE
Read neuroscope intervals function

### PR DESCRIPTION
**Pull request by GRVite**
Adding a function to read .evt files. It first searches if they are present in the nwb file. If the events are not in the nwb, it loads them from the .evt files.  I had to modify load_nwb_intervals to make it work (now it checks if the interval is among in the nwb file before trying to load it).

### CODE REVIEW

File: pynapple/io/neurosuite.py [hunk 1]
```diff
@@ -372,6 +372,35 @@ def load_lfp(self, filename=None, channel=None, extension='.eeg', frequency=1250
...
+        -------
+        IntervalSet  
+    
+        """
+        isets = self.load_nwb_intervals(name)
+        if str(type(isets)) !="<class 'pynapple.core.interval_set.IntervalSet'>":
+            print("reading intervals...")
+            evt_file = os.path.join(self.path, self.path.split('/')[-1] + "." + name + ".evt")
+            df = pd.read_csv(evt_file, delimiter=' ', usecols = [0], header = None)
+            start = nap.Ts(df.iloc[::2].values, time_units='ms')
```
Comment (gviejo) [inline pynapple/io/neurosuite.py:397]: It's too tedious. I would just reshape df.values with 2 columns and call intervalSet. 
Also it's possible that some files contains a start time, a middle time and an end time. I don't think we can assume it's always start and end.

File: pynapple/io/neurosuite.py [hunk 2]
```diff
@@ -372,6 +372,35 @@ def load_lfp(self, filename=None, channel=None, extension='.eeg', frequency=1250
...
+    
+        Returns
+        -------
+        IntervalSet  
+    
+        """
+        isets = self.load_nwb_intervals(name)
+        if str(type(isets)) !="<class 'pynapple.core.interval_set.IntervalSet'>":
+            print("reading intervals...")
+            evt_file = os.path.join(self.path, self.path.split('/')[-1] + "." + name + ".evt")
```
[the remark this answers is not in the document]
Comment (GRVite) [reply pynapple/io/neurosuite.py:395]: You actually call it like that, rem_ep = data.read_neuroscope_intervals('rem')
And yes, that is the name of the file. I will detail this in the description.

File: pynapple/io/neurosuite.py (hunk 1 above)
Comment (GRVite) [reply pynapple/io/neurosuite.py]: Changed

File: pynapple/io/neurosuite.py [hunk 3]
```diff
@@ -382,7 +382,10 @@ def read_neuroscope_intervals(self, name):
     
         Parameters
         ----------
-        name: name of the .evt file
+        name: name of the .evt file. The name of the file in the loading directory should
```
Comment (gviejo) [inline pynapple/io/neurosuite.py:385]: The numpy docstring format should be 
name: str
description

File: pynapple/io/neurosuite.py [hunk 4]
```diff
@@ -394,9 +397,8 @@ def read_neuroscope_intervals(self, name):
             print("reading intervals...")
```
Comment (gviejo) [inline pynapple/io/neurosuite.py:397]: Hum maybe not necessary :upside_down_face:

Comment (gviejo) [reply pynapple/io/neurosuite.py:397]: The print I mean :upside_down_face:

File: pynapple/io/neurosuite.py [hunk 5]
```diff
@@ -394,9 +397,8 @@ def read_neuroscope_intervals(self, name):
...
             evt_file = os.path.join(self.path, self.path.split('/')[-1] + "." + name + ".evt")
```
Comment (gviejo) [inline pynapple/io/neurosuite.py:398]: It should be self.basename and not dependant on the path

File: pynapple/io/neurosuite.py [hunk 6]
```diff
@@ -394,9 +397,8 @@ def read_neuroscope_intervals(self, name):
...
             df = pd.read_csv(evt_file, delimiter=' ', usecols = [0], header = None)
-            start = nap.Ts(df.iloc[::2].values, time_units='ms')
-            end = nap.Ts(df.iloc[1::2].values, time_units='ms')
-            isets = nap.IntervalSet(start.index, end.index, time_units='ms')
+            isets = nap.IntervalSet(df.iloc[::2].values, 
+                        df.iloc[1::2].values, time_units='ms')
             print(isets)
```
Comment (gviejo) [inline pynapple/io/neurosuite.py:402]: There is a print left.

File: pynapple/io/neurosuite.py (hunk 5 above)
Comment (gviejo) [reply pynapple/io/neurosuite.py:398]: I would do something like this : 
```
def read_neuroscope_intervals(self, pathtofile=None, name=None):
      if pathtofile is None and name is not None:
             pathtofile = os.path.join(self.path, self.basename + "." + name + ".evt")
     else:
            raise RuntimeError('Please specifiy a path or name")
```

Comment (GRVite) [reply pynapple/io/neurosuite.py:398]: Alright, (;

**Pull request closed without merging**